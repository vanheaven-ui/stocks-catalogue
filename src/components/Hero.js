import { Card } from 'react-bootstrap';
import hero from '../images/hero.jpg';
import styles from '../styles/hero.module.css';

const Hero = () => (
  <Card className="bg-dark text-white">
    <Card.Img src={hero} alt="Card image" />
    <Card.ImgOverlay>
      <Card.Title className={styles.cTitle}>
        INFORMATION FOR YOUR
        <br />
        NEXT STOCKS INVESTMENT
      </Card.Title>
    </Card.ImgOverlay>
  </Card>
);

export default Hero;
