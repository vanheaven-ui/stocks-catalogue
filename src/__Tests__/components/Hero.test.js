import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';
import Hero from '../../components/Hero';
import rootNode from '../helpers';
import '@testing-library/jest-dom';

describe('Hero component', () => {
  afterEach(cleanup);

  test('renders without crashing', () => {
    ReactDOM.render(<Hero />, rootNode());
  });

  test('doesnot change unexpectedly', () => {
    const tree = renderer.create(<Hero />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the right components', () => {
    const { getByText } = render(<Hero />);
    expect(getByText(/INFORMATION FOR YOUR/)).toBeInTheDocument();
  });
});
