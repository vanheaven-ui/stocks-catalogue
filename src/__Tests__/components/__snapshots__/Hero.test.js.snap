// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component doesnot change unexpectedly 1`] = `
<div
  className="bg-dark text-white card"
  style={
    Object {
      "height": 200,
    }
  }
>
  <img
    alt="Card image"
    className="card-img"
    src="hero.jpg"
  />
  <div
    className="card-img-overlay"
  >
    <div
      className="card-title h5"
      style={
        Object {
          "fontSize": 48,
          "textShadow": "3px 2px black",
        }
      }
    >
      INFORMATION FOR YOUR
      <br />
      NEXT STOCKS INVESTMENT
    </div>
  </div>
</div>
`;
